src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -421,7 +428,7 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return $this->name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
+        return 'app'.md5(get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
```

Comment:
Could this potentially break existing applications with multiple kernels?

Reply:
How? Each Kernel class must have a different FQCN anyway, so it's still unique.

Reply:
Having additional arguments in the application kernel class that are different for each instance does work too, doesn't it?

Reply:
It can work, but this won't break with my change as in this case, the cache cannot be shared between apps (today already).

Reply:
This will break apps rely on the kernel name to create "virtual" kernels. For example, I've some projects based on this idea https://github.com/yceruto/symfony-skeleton-vkernel.

I don't know whether this practice is right or not, but it simplify our projects sharing vendors, config and src. It work well thank to the kernel name and just one Kernel class.

Reply:
Then, you need to override the `getContainerClass()`.